### ISSUE TITLE
How to build SDL3 for emscripten?

### ISSUE THREAD

**Issue by slouken** [libsdl-org/SDL#11080]
Missing in the documentation for SDL3 is how to actually build an application for the web, using emscripten.

There’s the README-emscripten.md file in the docs folder, but it seems to be for SDL2, since it tells me to use the -sUSE_SDL=2 command. Building an SDL3-application with this command obviously won’t work, since the SDL version that emscripten contains is SDL2, which doesn’t have many of the SDL3-specific functions.

**Comment by icculus** [comment]
I'll take care of this. It's actually surprisingly like building for Linux, CMake and all.

**Comment by martinstarkov** [comment]
How are the SDL support libs (SDL_ttf, SDL_mixer, SDL_image) going to work with SDL3? Their release pages are all currently still on 2.x.x. Asking here because in my CMake for Emscripten I'm also using the ```--use-port=sdl2_ttf``` commands.
```cmake
  set(ECXXFLAGS "${ECXXFLAGS} -std=c++17 --use-port=sdl2 --use-port=sdl2_image:formats=bmp,png,xpm,jpg --use-port=sdl2_mixer --use-port=sdl2_ttf")
```
It would be great to see how those are different as well with SDL3.

**Comment by icculus** [comment]
Turns out the documentation on how to build from source was pretty robust, it just needed the "here's how to use the prebuilt packages" section removed, at least until prebuilt packages exist for SDL3.

Some other things got updated too.

**Comment by martinstarkov** [comment]
Cool! Thanks for the update 👍 Will wait for those to move to SDL3 then before upgrading

**Comment by adrian-javid** [comment]
> > How are the SDL support libs (SDL_ttf, SDL_mixer, SDL_image) going to work with SDL3? Their release pages are all currently still on 2.x.x.
> 
> These will be moving to SDL3 releases soon (what's in revision control has already been updated for SDL3). These will also need prebuilt packages for Emscripten, which we don't control, or to be built from source code.

@icculus, on `emcc` version 5.0.1, if I run `emcc --show-ports` I see `sdl3`, but not the support libraries for SDL3 such as `sdl3_image`. What is the status of getting Emscripten ports of the SDL3 support libraries?

**Comment by madebr** [comment]
> [@icculus](https://github.com/icculus), on `emcc` version 5.0.1, if I run `emcc --show-ports` I see `sdl3`, but not the support libraries for SDL3 such as `sdl3_image`. What is the status of getting Emscripten ports of the SDL3 support libraries?

This is done [by the Emscripten project themselve](https://github.com/emscripten-core/emscripten/tree/main/tools/ports).
There's [already a SDL3_Image pr here](https://github.com/emscripten-core/emscripten/pull/24634) but it looks it stalled. So I guess it's up to you to address the issues and create your own pr.

**Comment by martinstarkov** [comment]
I waited and the emscripten SDL3 satellite library ports seem to be taking a while so I wrote a quick script to build them from source (on Windows). It allows you to specify the versions as well. Might be useful to anybody stumbling upon this issue (maybe @adrian-javid?) and looking to move to SDL3:
https://gist.github.com/martinstarkov/78f4ae77441ed8c503f916b95edd74cd
Note: I pulled the script directly from my larger project, so it may require some filepath changes, but the core is there for anybody trying to do this themselves.
